lychee-lib/src/collector.rs

```diff
@@ -65,16 +86,24 @@ impl Collector {
             })
             .flatten();
 
+        // FIXME (Techassi): This error should be returned. But I don't know
+        // how we can return it inside the stream.
+        let basic_auth_extractor = BasicAuthExtractor::new(self.basic_auth_selectors).unwrap();
+
```

Comment:
This needs to be fixed. I'm unsure how tho. Maybe you have some input here @mre

Reply:
Yeah, that's a tricky one.
The challenge will be to find a single type that will be returned from that function. The Result + Future + Stream combination won't make that easy.

But I wonder why you construct the extractor in there?
Why not pass it into the collector instead?
Here's how you could do that:
https://github.com/Techassi/lychee/compare/feat/basic-auth-per-domain...lycheeverse:lychee:feat/basic-auth-per-domain-fixes?expand=1
If you want, you can directly merge that branch into yours; it should merge cleanly.

Reply:
Merged in 92522c2